docs/docs/recipes/integrate-logto/go-web.mdx

```diff
@@ -0,0 +1,347 @@
+/>
+
+### Add a route for handling sign-in requests
+
+```go
+//main.go
+func main() {
+	// ...
+
+	// Add a link to perform a sign-in request on the home page
+	router.GET("/", func(ctx *gin.Context) {
+		// ...
+		homePage := `<h1>Hello Logto</h1>` +
+			"<div>" + authState + "</div>" +
+			// Add link
+			`<div><a href="/sign-in">Sign In</a></div>`
+
+		ctx.Data(http.StatusOK, "text/html; charset=utf-8", []byte(homePage))
+	})
+
+	// Add a route for handling sign-in requests
+	router.GET("/sign-in", func(ctx *gin.Context) {
+		session := sessions.Default(ctx)
+		logtoClient := client.NewLogtoClient(
+			logtoConfig,
+			&SessionStorage{session: session},
+		)
+
+		// The sign-in request is handled by Logto.
+		// The user will be redirected to the Redirect URI on signed in.
+		signInUri, err := logtoClient.SignIn("http://localhost:8080/sign-in-callback")
+		if err != nil {
+			ctx.String(http.StatusInternalServerError, err.Error())
+			return
+		}
+
+		// Redirect the user to the Logto sign-in page.
+		ctx.Redirect(http.StatusTemporaryRedirect, signInUri)
+	})
+
+	// ...
+}
+```
+
+### Add a route for handling sign-in callback requests
+
+When the user signs in successfully on the Logto sign-in page, Logto will redirect the user to the Redirect URI.
+
+Since the Redirect URI is `http://localhost:8080/sign-in-callback`, we add the `/sign-in-callback` route to handle the callback after signing in.
+
+```go
+// main.go
+func main() {
+	// ...
+
+	// Add a route for handling sign-in callback requests
+	router.GET("/sign-in-callback", func(ctx *gin.Context) {
+		session := sessions.Default(ctx)
+		logtoClient := client.NewLogtoClient(
+			logtoConfig,
+			&SessionStorage{session: session},
+		)
+
+		// The sign-in callback request is handled by Logto
+		err := logtoClient.HandleSignInCallback(ctx.Request)
+		if err != nil {
+			ctx.String(http.StatusInternalServerError, err.Error())
+			return
+		}
+
+		// Jump to the page specified by the developer.
+		// This example takes the user back to the home page.
+		ctx.Redirect(http.StatusTemporaryRedirect, "/")
+	})
+
+	// ...
+}
+```
+
+## Sign out
+
+### Configure Post Sign-out Redirect URI
+
+<ConfigurePostSignOutRedirectUri
+  figureSrc={postSignOutRedirectUriFigure}
+  postSignOutRedirectUri="http://localhost:8080"
+/>
+
+This configuration enables the user to return to the home page after signing out.
+
+### Add a route for handling signing out requests
+
+```go
+//main.go
+func main() {
+	// ...
+
+	// Add a link to perform a sign-out request on the home page
+	router.GET("/", func(ctx *gin.Context) {
+		// ...
+		homePage := `<h1>Hello Logto</h1>` +
+			"<div>" + authState + "</div>" +
+			`<div><a href="/sign-in">Sign In</a></div>` +
+			// Add link
+			`<div><a href="/sign-out">Sign Out</a></div>`
+
+		ctx.Data(http.StatusOK, "text/html; charset=utf-8", []byte(homePage))
+	})
+
+	// Add a route for handling signing out requests
+	router.GET("/sign-out", func(ctx *gin.Context) {
```

Comment:
I'm not sure, but shouldn't the "sign-out" route expect a "POST" request?

Reply:
We click the sign-out link on the page, and the browser performs a `GET` request.